app/services/the_tvdb_service.rb

```diff
@@ -2,6 +2,37 @@ class TheTvdbService
   BASE_URL = 'https://api.thetvdb.com'.freeze
   API_KEY = ENV['THE_TVDB_API_KEY'].freeze
 
+  # daily or weekly
+  def import!(worker_type)
+    data = query_data(worker_type)
+    return if data.blank?
+
+    data['Anime'].each do |media_ids|
+      # set the media_id (this is for our database)
+      media_id = media_ids[0]
+      series_id, season_id = media_ids[1]['thetvdb'].split('/')
+
+      response = get(build_episode_path(series_id, season_id))
+
+      next if response.code == 404
+      raise 'TVDB Error' unless response.success?
+
+      response = JSON.parse(response.body)
```

Comment:
The JSON parsing and error handling should probably be in the `get` method or something

Reply:
I don't think it is possible, the next cannot be inside a separate method from the loop, and it has to be the first thing to execute (because response.success? checks for 2xx but we don't want to raise on a 404). Maybe yielding (but not sure how to set it up).